src/xpk/core/tests/integration/test_gcluster_a3ultra.py

```diff
@@ -0,0 +1,120 @@
+"""
+Copyright 2024 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from xpk.core.docker_manager import DockerManager
+from xpk.core.gcluster_manager import GclusterManager
+from xpk.core.blueprint.blueprint_generator import BlueprintGenerator
+import pytest
+import os
+import shutil
+
+ctk_gcloud_cfg = os.getenv("GCLOUD_CFG_PATH")
+project_id = os.getenv("PROJECT_ID")
+region = os.getenv("REGION")
+zone = os.getenv("ZONE")
+auth_cidr = os.getenv("AUTH_CIDR")
+cluster_name = os.getenv("CLUSTER_NAME")
+
+uploads_dir = "uploads"
+
+
+def prepare_test(docker_path: str, bp_path: str) -> None:
+  os.mkdir(docker_path)
+  os.mkdir(bp_path)
+
+
+@pytest.mark.skip(
+    reason=(
+        "This test requires A3 capacity, therefore it should not be run on each"
+        " build. Please invoke it manually if needed. "
+    )
+)
+def test_create_a3_mega_deployment():
```

Comment:
I think you can split it into two tests for creating and deploying. you can skip the deploying part only. It's good to have the blueprint/deployment creation part checked.